transactions/rbf.ts

```diff
@@ -0,0 +1,435 @@
+        }
+
+        tx.addInput({
+          txid: utxo.txid,
+          index: utxo.vout,
+          witnessUtxo: {
+            script: this.p2btc.script,
+            amount: BigInt(utxo.value),
+          },
+          redeemScript: this.p2btc.redeemScript,
+          witnessScript: this.p2btc.witnessScript,
+          sequence: 0xfffffffd,
+        });
+        inputsTotal += utxo.value;
+      } else {
+        // check if we can add change output
+        const txWithChange = tx.clone();
+        actualFee = change;
+        txWithChange.addOutputAddress(this.wallet.btcAddress, BigInt(Math.floor(change)), this.network);
+        const sizeWithChange = await this.getTxSize(txWithChange);
+        const newFeeWithChange = Math.max(sizeWithChange * desiredFeeRate, this.transaction.fees + sizeWithChange);
+
+        if (newFeeWithChange + 1000 < change) {
+          // add change output
+          actualFee = newFeeWithChange;
+          const actualChange = change - newFeeWithChange;
+          tx.addOutputAddress(this.wallet.btcAddress, BigInt(Math.floor(actualChange)), this.network);
+          outputsTotal += actualChange;
+        }
+
+        done = true;
+      }
+    }
+
+    const signedTransaction = await this.signTx(tx, isDummy, options);
+    signedTransaction.finalize();
+
+    return {
+      transaction: signedTransaction,
+      fee: actualFee,
+    };
+  };
+
+  private constructRecommendedFees = async (
+    lowerName: keyof RbfRecommendedFees,
+    lowerFeeRate: number,
+    higherName: keyof RbfRecommendedFees,
+    higherFeeRate: number,
+  ): Promise<RbfRecommendedFees> => {
+    const [lowerTx, higherTx] = await Promise.allSettled([
+      this.compileTransaction(lowerFeeRate, true),
+      this.compileTransaction(higherFeeRate, true),
+    ]);
+    return {
+      [lowerName]: {
+        fee: lowerTx.status === 'fulfilled' ? lowerTx.value.fee : undefined,
+        feeRate: lowerFeeRate,
+        enoughFunds: lowerTx.status === 'fulfilled',
+      },
+      [higherName]: {
+        fee: higherTx.status === 'fulfilled' ? higherTx.value.fee : undefined,
+        feeRate: higherFeeRate,
+        enoughFunds: higherTx.status === 'fulfilled',
+      },
+    };
+  };
+
+  getRbfRecommendedFees = async (mempoolFees: RecommendedFeeResponse): Promise<RbfRecommendedFees> => {
+    const { minimumRbfFeeRate } = getRbfTransactionSummary(this.transaction);
+
+    const { halfHourFee: medium, fastestFee: high } = mempoolFees;
+
+    if (minimumRbfFeeRate < medium) {
+      return this.constructRecommendedFees('medium', medium, 'high', high);
+    }
+
+    if (minimumRbfFeeRate < high) {
+      return this.constructRecommendedFees('high', high, 'higher', Math.ceil(high * 1.2));
+    }
+
+    return this.constructRecommendedFees(
+      'higher',
+      Math.ceil(minimumRbfFeeRate * 1.1),
+      'highest',
+      Math.ceil(minimumRbfFeeRate * 1.2),
+    );
+  };
+
+  getReplacementTransaction = async (options: O) => {
```

Comment:
The `options` param here requires to contain the `ledgerTransport` object, though it's not needed for regular account tx. Can we make it optional?
<img width="797" alt="Screenshot 2023-11-24 at 21 22 26" src="https://github.com/secretkeylabs/xverse-core/assets/36603049/6a61b9ef-1608-4c62-9aa8-7a1a2190ad26">

Reply:
It looks like it should require to contain `feeRate` property instead